powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/TokenBehavior.java

```diff
@@ -274,31 +276,26 @@ public ValidateTokenResponse validateToken(ValidateTokenRequest request) throws
 
             // Check if the activation is in correct state
             final ActivationRecordEntity activation = token.getActivation();
+            final byte[] tokenSecret = BaseEncoding.base64().decode(token.getTokenSecret());
+            final boolean isTokenValid;
             if (!ActivationStatus.ACTIVE.equals(activation.getActivationStatus())) {
                 logger.info("Activation is not ACTIVE, activation ID: {}", activation.getActivationId());
-                // Rollback is not required, database is not used for writing
-                throw localizationProvider.buildExceptionForCode(ServiceError.ACTIVATION_INCORRECT_STATE);
-            }
-
-            final byte[] tokenSecret = BaseEncoding.base64().decode(token.getTokenSecret());
-
-            final boolean isTokenValid = tokenVerifier.validateTokenDigest(nonce, timestamp, tokenSecret, tokenDigest);
-
-            if (isTokenValid) {
-                final ValidateTokenResponse response = new ValidateTokenResponse();
-                response.setTokenValid(true);
-                response.setActivationId(activation.getActivationId());
-                response.setApplicationId(activation.getApplication().getId());
-                response.getApplicationRoles().addAll(activation.getApplication().getRoles());
-                response.getActivationFlags().addAll(activation.getFlags());
-                response.setUserId(activation.getUserId());
-                response.setSignatureType(signatureTypeConverter.convertFrom(token.getSignatureTypeCreated()));
-                return response;
+                isTokenValid = false;
             } else {
-                final ValidateTokenResponse response = new ValidateTokenResponse();
-                response.setTokenValid(false);
-                return response;
+                isTokenValid = tokenVerifier.validateTokenDigest(nonce, timestamp, tokenSecret, tokenDigest);
             }
+
+            final ValidateTokenResponse response = new ValidateTokenResponse();
+            response.setTokenValid(isTokenValid);
+            response.setActivationStatus(activationStatusConverter.convert(ActivationStatus.ACTIVE));
```

Comment:
Do I understand it correctly, that if the activation is not active, we still return activation status `ACTIVE` and `tokenValid=false`? Is that desired?

`!ActivationStatus.ACTIVE.equals(activation.getActivationStatus())`

Reply:
No, it's incorrect, fixed.